MFVNeutralino/python/NtupleCommon.py

```diff
@@ -268,16 +301,30 @@ def miniaod_ntuple_process(settings):
     process.mfvEvent.gen_particles_src = 'prunedGenParticles' # no idea if this lets gen_bquarks, gen_leptons work--may want the packed ones that have status 1 particles
     process.mfvEvent.gen_jets_src = 'slimmedGenJets'
     process.mfvEvent.pileup_info_src = 'slimmedAddPileupInfo'
-    process.mfvEvent.met_src = 'slimmedMETs'
+    # use corrected MET
+    #need to be changed to use TriggerFloats corrected MET
+    process.mfvEvent.met_src = cms.InputTag('slimmedMETs', '', 'Ntuple')
+
+    # MET correction and filters
+    # https://twiki.cern.ch/twiki/bin/view/CMS/MissingETUncertaintyPrescription#PF_MET
+    from PhysicsTools.PatUtils.tools.runMETCorrectionsAndUncertainties import runMetCorAndUncFromMiniAOD
+    process.load("Configuration.StandardSequences.GeometryRecoDB_cff") 
+    runMetCorAndUncFromMiniAOD(process,
+                               isData = not settings.is_mc,
+                               )
```

Comment:
Is it safe for us to include these MET filters for the non-MET-trigger events (meaning leptons/HT/bjet triggered)? I don't remember e.g. if events are vetoed by the filter. If so, we should make sure to only apply it when relevant.

Reply:
I think all those processes are just MET corrections and MET filter results. So those things won't reject events but just adding additional information into the event. Do you think it's ok to keep them for non-MET case (since it can be useless)? I think it won't have additional effects.

Reply:
Okay! I haven't looked into the implementation of the MET filters very closely, so wasn't sure if they were only adding the information in, or if cuts were being applied at this stage too. Definitely fine to add the info in regardless of trigger, as long as we remember not to cut out any events via these filters in the non-MET cases!